src/main/java/com/iridium/iridiumskyblock/gui/IslandBankGUI.java

```diff
@@ -54,6 +55,20 @@ public void onInventoryClick(InventoryClickEvent event) {
         Optional<BankItem> bankItem = IridiumSkyblock.getInstance().getBankItemList().stream().filter(item -> item.getItem().slot == event.getSlot()).findFirst();
         if (!bankItem.isPresent()) return;
 
+        if (event.isShiftClick()) {
+            switch (event.getClick()) {
+                case LEFT:
+                case SHIFT_LEFT:
+                    new IslandCustomAmountBankGUI().openInventory((Player) event.getWhoClicked(), this.getIsland(), bankItem.get(), false);
+                    break;
+                case RIGHT:
+                case SHIFT_RIGHT:
+                    new IslandCustomAmountBankGUI().openInventory((Player) event.getWhoClicked(), this.getIsland(), bankItem.get(), true);
+                    break;
+            }
```

Comment:
Can be shortened
```suggestion
            new IslandCustomAmountBankGUI().openInventory((Player) event.getWhoClicked(), this.getIsland(), bankItem.get(), event.isRightClick());
```

Reply:
Also I think shift click should remain as deposit/withdraw all